From c57c213009c2e6c9dd002c6693e8ac849b6b6f15 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:15 -0400
Subject: [tools] perf/machine: Don't open code assign dso->short_name

Message-id: <1408953210-26343-128-git-send-email-jolsa@redhat.com>
Patchwork-id: 88909
O-Subject: [PATCH RHEL7.1 BZ1133083 127/322] perf machine: Don't open code assign dso->short_name
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 7521ab592550d9e6542a496bcea11b40900690da
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Dec 10 10:58:47 2013 -0300

description
===========
Use dso__set_short_name instead, as it will release any previously,
possibly allocated, short name.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/machine.c b/tools/perf/util/machine.c
index f66f309..f85da9a 100644
--- a/tools/perf/util/machine.c
+++ b/tools/perf/util/machine.c
@@ -935,7 +935,7 @@ static int machine__process_kernel_mmap_event(struct machine *machine,
   if (name == NULL)
    goto out_problem;
 
-  map->dso->short_name = name;
+  dso__set_short_name(map->dso, name);
   map->dso->short_name_allocated = 1;
   map->end = map->start + event->mmap.len;
  } else if (is_kernel_mmap) {
-- 
1.7.1